core/src/main/scala/kafka/log/Log.scala

```diff
@@ -557,8 +587,8 @@ class Log(@volatile var dir: File,
     var entry = segments.floorEntry(startOffset)
 
     // attempt to read beyond the log end offset is an error
-    if(startOffset > next || entry == null)
-      throw new OffsetOutOfRangeException("Request for offset %d but we only have log segments in the range %d to %d.".format(startOffset, segments.firstKey, next))
+    if(startOffset > next || entry == null || startOffset < logStartOffset)
+      throw new OffsetOutOfRangeException("Request for offset %d but we only have log segments in the range %d to %d.".format(logStartOffset, logEndOffset, next))
```

Comment:
I think the original error message is wrong. Why the range of log segments depend on an offset provided by the user?